js/errors.ts

```diff
@@ -1,17 +1,96 @@
 import { deno as fbs } from "gen/msg_generated";
 
-// @internal
 export class DenoError<T extends fbs.ErrorKind> extends Error {
   constructor(readonly kind: T, msg: string) {
     super(msg);
-    this.name = `deno.${fbs.ErrorKind[kind]}`;
+    this.name = `deno.Err${fbs.ErrorKind[kind]}`;
   }
 }
 
+const errorClasses = new Map();
+
+function ErrorFactory<T extends fbs.ErrorKind>(
+  kind: T
+): new (msg: string) => DenoError<T> {
+  const name = `Err${fbs.ErrorKind[kind]}`;
+  const anonymousClass = class extends DenoError<T> {
+    constructor(msg: string) {
+      super(kind, msg);
+    }
+  };
+  Object.defineProperty(anonymousClass, "name", {
```

Comment:
`writable`, `configurable` and `enumerable` are all `false` by default.  I would recommend being explicit about all of them (or none of them).

Reply:
@kitsonk Thanks for the comment